### DISCUSSION TITLE
使用rec训练反向180°的图像，是否需要将图像转为0°方向训练

### DISCUSSION THREAD

**Discussion by alanxinn** [PaddlePaddle/PaddleOCR#14188] [Q&A]
图像文字为180°，训练rec模型时，是否需要将图像翻转为文字方向为0°，或者我的训练的图像都保证为180°

**Comment by GreatV** [comment]
需要考虑包含和不包含方向分类器两种情况：

### 1. **情况一：推理阶段包含方向分类器**
- **推理流程**：当推理时使用方向分类器，图像会被自动旋转调整为标准的0°方向。因此，文本识别模型接收到的输入图像始终是0°方向的。
- **训练数据准备**：
  - **图像方向**：将所有训练图像调整为标准的0°方向。这样，模型会专门学习如何识别0°方向的文字，与推理阶段的调整相匹配。
  - **数据增强**：可以适当进行数据增强（如轻微的旋转、缩放、噪声等），但不要加入180°翻转的数据，因为方向分类器会确保图像被调整为0°方向。

### 2. **情况二：推理阶段不包含方向分类器**
- **推理流程**：如果不使用方向分类器，图像将直接输入文本识别模型，而图像可能包含不同的角度（如0°、180°等）。
- **训练数据准备**：
  - **图像方向**：你需要在训练集中包含各种可能的图像方向。例如，如果你的应用场景中常见180°的文字图像，可以在训练数据中加入这些180°的样本。
  - **数据增强**：强烈建议使用数据增强来生成不同方向的图像（包括0°、90°、180°、270°）。这将增加模型的鲁棒性，使其能够应对各种文字方向。

**Reply by alanxinn** [reply]
> 需要考虑包含和不包含方向分类器两种情况：
> 
> ### 1. **情况一：推理阶段包含方向分类器**
> * **推理流程**：当推理时使用方向分类器，图像会被自动旋转调整为标准的0°方向。因此，文本识别模型接收到的输入图像始终是0°方向的。
> * **训练数据准备**：
>   
>   * **图像方向**：将所有训练图像调整为标准的0°方向。这样，模型会专门学习如何识别0°方向的文字，与推理阶段的调整相匹配。
>   * **数据增强**：可以适当进行数据增强（如轻微的旋转、缩放、噪声等），但不要加入180°翻转的数据，因为方向分类器会确保图像被调整为0°方向。
> 
> ### 2. **情况二：推理阶段不包含方向分类器**
> * **推理流程**：如果不使用方向分类器，图像将直接输入文本识别模型，而图像可能包含不同的角度（如0°、180°等）。
> * **训练数据准备**：
>   
>   * **图像方向**：你需要在训练集中包含各种可能的图像方向。例如，如果你的应用场景中常见180°的文字图像，可以在训练数据中加入这些180°的样本。
>   * **数据增强**：强烈建议使用数据增强来生成不同方向的图像（包括0°、90°、180°、270°）。这将增加模型的鲁棒性，使其能够应对各种文字方向。

感谢回答
